connection-limit/connection_limit_kern.c

```diff
@@ -273,17 +273,20 @@ int _xdp_limit_conn(struct xdp_md *ctx)
     if (data + sizeof(struct ethhdr)+ 1 > data_end)
         return XDP_PASS;
 
-    /* Check if its valid ip packet */
+    
     struct iphdr *iph = (struct iphdr *)(data + sizeof(struct ethhdr));
-    if (iph + 1 > data_end)
+    __u8 l4_offset = iph->ihl * 4; // ip header length
```

Comment:
Endianness is handled in the definition of 'struct iphdr'. However please have a test-case to verify that the value for IHL is read correctly from the packet.

Reply:
What sort of check did you have in mind?  I am not sure one is necessary.